perfkitbenchmarker/providers/gcp/gcp_dpb_dataflow_template.py

```diff
@@ -0,0 +1,153 @@
+
+  def __init__(self, dpb_service_spec):
+    super(GcpDpbDataflowTemplate, self).__init__(dpb_service_spec)
+    self.project = util.GetDefaultProject()
+    self.input_sub_empty = False
+    self.job_drained = False
+
+  @staticmethod
+  def CheckPrerequisites(benchmark_config):
+    del benchmark_config  # Unused
+    if not FLAGS.dpb_df_template_gcs_location:
+      raise errors.Config.InvalidValue('Template GCS location missing.')
+
+  def Create(self):
+    """See base class."""
+    pass
+
+  def Delete(self):
+    """See base class."""
+    pass
+
+  def SubmitJob(
+      self,
+      template_gcs_location=None,
+      job_poll_interval=None,
+      job_arguments=None,
+      job_input_sub = None):
+
+    worker_machine_type = self.spec.worker_group.vm_spec.machine_type
+    num_workers = self.spec.worker_count
+    max_workers = self.spec.worker_count
+
+    now = datetime.datetime.now()
+    job_name = template_gcs_location.split('/')[-1] \
+            + '_' + now.strftime("%Y%m%d_%H%M%S")
+    region = util.GetRegionFromZone(FLAGS.dpb_service_zone)
+
+    cmd = util.GcloudCommand(self, 'dataflow', 'jobs', 'run', job_name)
+    cmd.flags = {
+        'project': self.project,
+        'gcs-location': template_gcs_location,
+        'staging-location': FLAGS.dpb_dataflow_temp_location,
+        'region': region,
+        'worker-region': region,
+        'worker-machine-type': worker_machine_type,
+        'num-workers': num_workers,
+        'max-workers': max_workers,
+        'parameters': ','.join(job_arguments),
+        'format': 'json',
+    }
+
+    stdout, _, _ = cmd.Issue()
+
+    # Parse output to retrieve submitted job ID
+    try:
+      result = json.loads(stdout)
+      self.job_id = result['id']
+    except Exception as err:
+      logging.error('Failed to parse Dataflow job ID: {}'.format(err))
+      raise
+
+    logging.info('Dataflow job ID: %s', self.job_id)
+    # TODO: return JobResult() with pre-computed time stats
+    return self._WaitForJob(
+        job_input_sub, FLAGS.dpb_dataflow_timeout, job_poll_interval)
+
+  def _GetCompletedJob(self, job_id):
+    """See base class."""
+    job_input_sub = job_id
+
+    # Job completed if input subscription is empty *and* job is drained;
+    # otherwise keep waiting
+    if not self.input_sub_empty:
+      backlog_size = self.GetSubscriptionBacklogSize(job_input_sub)
+      logging.info('Polling: Backlog size of subscription {} is {}'
+          .format(job_input_sub, backlog_size))
+      if backlog_size == 0:
+        self.input_sub_empty = True
+        # Start draining job once input subscription is empty
+        cmd = util.GcloudCommand(
+            self, 'dataflow', 'jobs', 'drain', self.job_id)
+        cmd.flags = {
+            'project': self.project,
+            'region': util.GetRegionFromZone(FLAGS.dpb_service_zone),
+            'format': 'json',
+        }
+        logging.info('Polling: Draining job {} ...'.format(self.job_id))
```

Comment:
Don't interpolate variables directly in strings to be logged. Refer to https://google.github.io/styleguide/pyguide.html#3101-logging